File: ReSharper.FSharp/src/FSharp.Psi/src/Impl/Cache2/ZoneMarker.cs
using JetBrains.Application.BuildScript.Application.Zones;
using JetBrains.RdBackend.Common.Env;
using JetBrains.Rider.Backend.Env;

namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Impl.Cache2
{
  [ZoneMarker]
  public class ZoneMarker : IRequire<IResharperHostCoreFeatureZone>, IRequire<IRiderFeatureEnvironmentZone>
  {
  }
}

File: ReSharper.FSharp/src/FSharp/FSharp.Tests.Host/ZoneMarker.fs
namespace JetBrains.ReSharper.Plugins.FSharp.TestsHost

open JetBrains.Application.BuildScript.Application.Zones
open JetBrains.Rider.Backend.Env

[<ZoneMarker>]
type ZoneMarker() =
    interface IRequire<IRiderCoreTestingZone>